src/types.rs

```diff
@@ -262,9 +263,13 @@ pub type RecordRow = RecordRowF<Box<Types>>;
 #[derive(Clone, PartialEq, Debug)]
 /// Concrete, recursive definition for record rows.
 pub struct RecordRows(pub RecordRowsF<Box<Types>, Box<RecordRows>>);
+
 /// Concrete, recursive type for a Nickel type.
 #[derive(Clone, PartialEq, Debug)]
-pub struct Types(pub TypeF<Box<Types>, RecordRows, EnumRows>);
+pub struct Types {
+    pub ty: TypeF<Box<Types>, RecordRows, EnumRows>,
```

Comment:
I must say I'm not a fan of something as short as `ty` for a field name. I don't have a great suggestion, unfortunately. An obvious one is `types` (or maybe one day we should use `typ` as proposed by Julien? But let's see that later, all at once). Could also be `inner`? `typef`? `node`?